1.1-Introduction-to-Schnorr.ipynb

```diff
@@ -171,29 +171,38 @@
    "outputs": [],
    "source": [
     "msg = hashlib.sha256(b'message').digest()\n",
-    "d = ECKey().generate()\n",
-    "P = d.get_pubkey()\n",
     "\n",
-    "k = ECKey().generate()\n",
-    "R = k.get_pubkey()\n",
+    "# Generate a private/public key pair\n",
+    "d, P = generate_key_pair()\n",
+    "\n",
+    "# Generate a nonce scalar and associated nonce point\n",
+    "k, R = generate_key_pair()\n",
     "\n",
     "# Check that nonce is quadratic residue modulo the field size\n",
     "# If not, negate k\n",
     "# Method: jacobi_symbol(R.get_y(), SECP256K1_FIELD_SIZE)\n",
     "if  # TODO: implement\n",
     "    k.negate()\n",
+    "# Note that there is no need to negate R, since we only use the x value of R below\n",
     "\n",
     "# Generate s = k + sha256(R_x|P|msg) * d\n",
     "# Method: hashlib.sha256(bytes).digest() will give you the byte digest of the SHA256 of some bytes\n",
     "# Turn that digest into a ECKey object called h, and then set s = k + h * d\n",
+    "R_x_bytes = R.get_x().to_bytes(32, 'big')\n",
+    "P_bytes = P.get_bytes()\n",
     "h_bytes =  # TODO: implement\n",
-    "h =  # TODO: implement\n",
+    "h = ECKey().set(h_bytes)\n",
```

Comment:
you left the answer here.

Reply:
I did this deliberately. I think the exercise should be setting the `h_bytes` and then we convert that to a `ECKey()`.